snapcraft/tests/test_sources.py

```diff
@@ -325,3 +334,22 @@ def test_get_tar_source_from_uri(self):
             with self.subTest(key=source):
                 self.assertEqual(
                     snapcraft.sources._get_source_type_from_uri(source), 'tar')
+
+    @unittest.mock.patch('snapcraft.sources.Git.pull')
+    def test_get_git_source_from_uri(self, mock_pull):
+        sources = [
+            'git://github.com:ubuntu-core/snapcraft.git',
+            'git@github.com:ubuntu-core/snapcraft.git'
+        ]
+
+        for source in sources:
+            with self.subTest(key=source):
+
+                options = MockOptions(source=source,source_type='git')
+                local = snapcraft.sources
+                local.get(
+                    sourcedir = 'testsourcedir',
+                    builddir = 'testbuilddir',
```

Comment:
since now the sourcedir and builddir are not used in the test, a good practices is to make it explicit that they are dummies (http://xunitpatterns.com/Dummy%20Object.html)
So, please change them like this:
sourcedir = 'dummy',
builddir = 'dummy'